llvm/lib/Support/APInt.cpp

```diff
@@ -2105,7 +2100,7 @@ void APInt::fromString(unsigned numbits, StringRef str, uint8_t radix) {
   if (isSingleWord())
     U.VAL = 0;
   else
-    U.pVal = getClearedMemory(getNumWords());
+    U.pVal = new WordType[getNumWords()]();
```

Comment:
This one looks completely wrong, because the following code relies on a full zero initialization.

Reply:
`new T[]()` (as opposed to `new T[]`) will default-initialize to zero?